### ISSUE TITLE
Controller default function must be called explicitly...

### ISSUE THREAD

**Issue by RobertAtomic** [datafolklabs/cement#421]
Hi, 

I'm new to Cement, so please be merciful if this is user error...

After creating a new app with boss my application controllers do not call their default functions unless they're specified on the command line. Additionally, the functions exported with hide=True are not hidden.  This behavior can be seen by simply creating a new app with boss, but here'e my controller anyway...

```
"""myapp base controller."""

from cement.ext.ext_argparse import ArgparseController, expose

class myappBaseController(ArgparseController):
    class Meta:
        label = 'base'
        description = 'My Project'
        arguments = [ 
            (['-f', '--foo'],
             dict(help='the notorious foo option', dest='foo', action='store',
                  metavar='TEXT') ),
            ]   

    @expose(hide=True)
    def default(self):
        print("Inside myappBaseController.default().")
```

and here is the behavior:
```
(env) :~/myapp/myapp/cli/controllers$ myapp 
usage: myapp [-h] [--debug] [--quiet] [-f TEXT] {default} ...
myapp: error: too few arguments
```
```
(env) :~/myapp/myapp/cli/controllers$ myapp default
Inside myappBaseController.default().
```

I'd like to be able to just call
```
myapp
```
and have it run the default function.

Thanks for any help!

**Comment by akhilman** [comment]
Can not reproduce with `cement-script` template. Default function runs as expected without passing subcommand.

**Comment by derks** [comment]
@RobertAtomic what version of Python are you using?  There is a usage bug in Arparse < 3.5 that doesn't allow support for the default function in Cement. I believe this is documented but will have to look (Im currently AFK).

**Comment by jtpereyda** [comment]
@derks I tried @RobertAtomic code using Python 4.4 and default seems to work just fine.

Code:

```python
from cement.core.foundation import CementApp
from cement.ext.ext_argparse import ArgparseController, expose

class myappBaseController(ArgparseController):
    class Meta:
        label = 'base'
        description = 'My Project'
        arguments = [
            (['-f', '--foo'],
             dict(help='the notorious foo option', dest='foo', action='store',
                  metavar='TEXT') ),
            ]

    @expose(hide=True)
    def default(self):
        self.app.log.info("Inside myappBaseController.default().")

class MyApp(CementApp):
    class Meta:
        label = 'demo'
        base_controller = 'base'
        handlers = [myappBaseController]

with MyApp() as app:
    app.run()
```

Output:
```
> python3 --version
Python 3.4.4

> python3 cement-issue-421.py --help
usage: demo [-h] [--debug] [--quiet] [-f TEXT] {default} ...

My Project

optional arguments:
  -h, --help           show this help message and exit
  --debug              toggle debug output
  --quiet              suppress all output
  -f TEXT, --foo TEXT  the notorious foo option

sub-commands:
  {default}

> python3 cement-issue-421.py
INFO: Inside myappBaseController.default().
```

The docs [say](http://builtoncement.com/2.10/api/ext/ext_argparse.html#cement-ext-ext-argparse) that default is only available in Python 3.4+, but perhaps this is no longer the case.

**Comment by derks** [comment]
@akhilman and @jtpereyda thank you for testing that out.  @RobertAtomic as mentioned, this is not an issue with Cement, it is actually how Argparse functions.  In Python < 3.4 Argparse will throw a `too few arguments` exception... which makes it impossible to support the `default` functionality in Cement.

Python 3.4+ will do what you want.
